### PR TITLE
mvapich2: Adding external find support.

### PR DESCRIPTION
**Pull request by rblake-llnl**
This picks up all the options that are currently settable by
the spack package. It also detects the compiler and sets it
appropriately.

### CODE REVIEW

**Comment by adamjstewart** [comment]
@nithintsk @harisubramoni can you review this PR? The changes in this PR allows users to run `spack external find mvapich2` and have Spack automatically populate `packages.yaml` with external mvapich2 installations.

File: var/spack/repos/builtin/packages/mvapich2/package.py [hunk 1]
```diff
@@ -122,6 +125,108 @@ class Mvapich2(AutotoolsPackage):
...
+                variants += '~cuda'
+
+            if re.search('--enable-registration-cache', output):
+                variants += '+regcache'
+            else:
+                variants += '~regcache'
+
+            match = re.search(r'--enable-threads=(\S+)', output)
+            if match:
+                variants += " +threads=" + match.group(1)
```
Comment (nithintsk) [inline var/spack/repos/builtin/packages/mvapich2/package.py]: The + symbols next to threads can be avoided, I believe.

File: var/spack/repos/builtin/packages/mvapich2/package.py [hunk 2]
```diff
@@ -122,6 +125,108 @@ class Mvapich2(AutotoolsPackage):
...
+            if re.search(r'--with-pm=[A-Za-z0-9:]*gforker', output):
+                pms.append('gforker')
+            if re.search(r'--with-pm=[A-Za-z0-9:]*remshell', output):
+                pms.append('remshell')
+            if pms:
+                variants += " process_managers=" + ",".join(pms)
+
+            fabrics = {
+                'sock': 'ch3:sock',
+                'nemesistcpib': 'ch3:nemesis:tcp:ib',
```
Comment (nithintsk) [inline var/spack/repos/builtin/packages/mvapich2/package.py]: The last colon for the below two symbols might have to be a comma, when comparing with the current package.py file
```
ch3:nemesis:tcp:ib --> ch3:nemesis:tcp,ib
ch3:nemesis:ib:tcp --> ch3:nemesis:ib,tcp
```
```
        elif 'fabrics=nemesistcpib' in self.spec:
            opts = ["--with-device=ch3:nemesis:tcp,ib"]
        elif 'fabrics=nemesisibtcp' in self.spec:
            opts = ["--with-device=ch3:nemesis:ib,tcp"]
```

**Review by nithintsk — CHANGES_REQUESTED** [review]
I have noticed some details that might need edits. Do let me know what you think.

**Comment by rblake-llnl** [comment]
@nithintsk Added your changes.  Thanks!

**Review by nithintsk — APPROVED** [review]
Thanks for making the changes. Looks good to me.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**